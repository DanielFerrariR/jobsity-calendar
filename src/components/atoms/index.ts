export { default as Box } from './box'
export { default as Button } from './button'
export { default as Paper } from './paper'
export { default as Typography } from './typography'
export { default as TextField } from './text_field'
export { default as Container } from './container'
export { default as AppBar } from './appbar'
export { default as IconButton } from './icon_button'
export { default as Toolbar } from './toolbar'
export { default as Image } from './image'
export { default as Grid } from './grid'
export { default as Link } from './link'
export { default as Checkbox } from './checkbox'
export { default as Accordion } from './accordion'
export { default as AccordionSummary } from './accordion_summary'
export { default as AccordionDetails } from './accordion_details'
export { default as Tooltip } from './tooltip'
export { default as Modal } from './modal'
export { default as MenuItem } from './menu_item'
export { default as ListItemIcon } from './list_item_icon'
export { default as ListItemText } from './list_item_text'
export { default as ListItemSecondaryAction } from './list_item_secondary_action'
export { default as MenuList } from './menu_list'
export { default as Popover } from './popover'
export { default as Fade } from './fade'
export { default as Backdrop } from './backdrop'
export { default as CircularProgress } from './circular_progress'
export { default as LinearProgress } from './linear_progress'
export { default as Skeleton } from './skeleton'
export { default as List } from './list'
export { default as ListItem } from './list_item'
export { default as ClickAwayListener } from './click_away_listener'
export { default as Badge } from './badge'
export { default as Portal } from './portal'
export { default as Snackbar } from './snackbar'
export { default as Step } from './step'
export { default as StepContent } from './stepContent'
export { default as StepLabel } from './stepLabel'
export { default as Stepper } from './stepper'
export { default as TableBody } from './table_body'
export { default as TableContainer } from './table_container'
export { default as TableHead } from './table_head'
export { default as TableRow } from './table_row'
export { default as Table } from './table'
export { default as TableCell } from './table_cell'
export { default as DateTimePicker } from './date_time_picker'
