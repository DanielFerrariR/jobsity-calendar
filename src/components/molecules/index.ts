export { CustomTableHead, CustomTableBody } from './custom_table'
export { default as Modal } from './modal'
export { default as ReminderCard } from './reminder_card'
