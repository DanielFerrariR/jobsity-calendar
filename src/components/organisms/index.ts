export { default as Calendar } from './calendar'
export { default as ListRemindersModal } from './list_reminders_modal'
export { default as CreateReminderModal } from './create_reminder_modal'
export { default as EditReminderModal } from './edit_reminder_modal'
