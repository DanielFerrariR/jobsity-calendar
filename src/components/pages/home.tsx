import React from 'react'
import { Container } from 'src/components/atoms'
import { Calendar } from 'src/components/organisms'

const Home: React.FC = () => {
  return (
    <Container mt={4}>
      <Calendar />
    </Container>
  )
}

export default Home
