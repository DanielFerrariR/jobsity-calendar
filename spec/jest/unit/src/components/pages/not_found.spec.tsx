import React from 'react'
import { NotFound } from 'src/components/pages'
import { render } from 'spec/jest/config/test_utils'
import { axe } from 'jest-axe'

describe('testing NotFound component', () => {
  test('should snapshot works with english language', () => {
    const { container } = render(<NotFound />)

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with portuguese language', () => {
    const { container } = render(<NotFound />, {
      language: 'pt'
    })

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with spanish language', () => {
    const { container } = render(<NotFound />, {
      language: 'es'
    })

    expect(container).toMatchSnapshot()
  })

  test('should the page be accessible', async () => {
    const { container } = render(<NotFound />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
