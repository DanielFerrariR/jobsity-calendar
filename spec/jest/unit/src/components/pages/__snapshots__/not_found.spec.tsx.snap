// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing NotFound component should snapshot works with english language 1`] = `
<div>
  <div
    class="MuiContainer-root Component-root Component-root MuiContainer-maxWidthLg"
  >
    <h4
      class="MuiTypography-root Component-root Component-root MuiTypography-h4 MuiTypography-alignCenter"
    >
      Sorry, we couldn't find this page.
    </h4>
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
    >
      The link you followed may be broken or the page may have been removed.
    </p>
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover Component-root Component-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
      data-testid="login-link"
      href="/front/login"
    >
      Back
    </a>
  </div>
</div>
`;

exports[`testing NotFound component should snapshot works with portuguese language 1`] = `
<div>
  <div
    class="MuiContainer-root Component-root Component-root MuiContainer-maxWidthLg"
  >
    <h4
      class="MuiTypography-root Component-root Component-root MuiTypography-h4 MuiTypography-alignCenter"
    >
      Desculpe, não conseguimos encontrar essa página.
    </h4>
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
    >
      O link que você seguiu pode estar quebrado ou a página pode ter sido removida.
    </p>
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover Component-root Component-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
      data-testid="login-link"
      href="/front/login"
    >
      Voltar
    </a>
  </div>
</div>
`;

exports[`testing NotFound component should snapshot works with spanish language 1`] = `
<div>
  <div
    class="MuiContainer-root Component-root Component-root MuiContainer-maxWidthLg"
  >
    <h4
      class="MuiTypography-root Component-root Component-root MuiTypography-h4 MuiTypography-alignCenter"
    >
      Lo sentimos, no pudimos encontrar esta página.
    </h4>
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
    >
      El enlace que siguió puede estar roto o la página puede haber sido eliminada.
    </p>
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover Component-root Component-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
      data-testid="login-link"
      href="/front/login"
    >
      Volver
    </a>
  </div>
</div>
`;
